### PR TITLE
Shipping Labels: Add "activate service packages" form.

### LINKED ISSUE #3992: Shipping labels: activate new service carrier packages
In the shipping label creation form, allow the user to add new service carrier packages:

Screen Shot 2021-05-07 at 12 10 52
Screen Shot 2021-05-07 at 12 17 39

Please note that as per discussion with @Garance91540, we will use radio-buttons instead of checkboxes in the package selector list, to clearly highlight that it's single choice list.

### PR DESCRIPTION
**Pull request by hafizrahman**
# What this PR does

This PR adds a functionality to allow users to activate a service package (also known as predefined package), during shipping label creation process.

Fixes #3992

# Testing Instructions
0. Make sure the test site has Shipping Labels creation enabled, and have an Order where Shipping Labels can be created.
1. Delete all activated packages in wp-admin: `wp-admin/admin.php?page=wc-settings&tab=shipping&section=woocommerce-services-settings`.
2. In the app, find the Order from step 0 and create a Shipping Label, complete the "Ship from" and "Ship to" steps, then continue with the "Packaging details" step.
3. Tap the "Package selected" spinner, this will open the "Add new package" screen. 
4. Switch to the "Service Package" tab. This will show a loading screen, and then finally show a list of active packages that can be activated.
5. Select a package (remember the category and name), tap Done.
6. If the package is saved properly,  the screen will be navigated back to the "Package Details" screen. Make sure the name of the recently activated package is shown in the "Package selected" spinner.
7. Tap the "Package selected" spinner. Make sure this opens the "Selected Package" screen, and the recently added package is listed there.
8. Tap the "Add New Package" button again. Open the "Service Package" tab and make sure that the recently activated package is no longer listed there.
9. Pick another package there, and tap "Done". This will again navigate back to the "Package Details" screen. Make sure the package name is now shown in the "Package selected" spinner.

See video below for the entire flow:

https://user-images.githubusercontent.com/266376/126443151-cce449f6-3147-4495-922d-a3d2d497886a.mp4

# Known limitations
- No unit tests yet. This will be added in a separate PR.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelCreateServicePackageAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                        ListItem.Package(shippingPackage, isSelected)
+                    }
+                )
+                list
+            }
+    }
+
+    private sealed class ListItem {
+        data class Header(val title: String) : ListItem()
+        data class Package(val data: ShippingPackage, val isChecked: Boolean = false) : ListItem()
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelCreateServicePackageAdapter.kt]: I would suggest that you change the `data` type to `ServicePackageUiModel` instead, and remove the `isChecked` property, as it's just a duplicate.
This way, you will be able to simplify the logic of `updateData`.

Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelCreateServicePackageAdapter.kt]: Thanks, that's a good idea.

**Comment by hafizrahman** [comment]
This PR still has one case to cover (what to show when all packages have been selected and there's no selectable packages anymore). Since Peril bot has mentioned that this PR is already large, I'm splitting that to a separate issue #4458.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelServicePackageAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    fun updateData(uiModels: List<ServicePackageUiModel>) {
+        val items = uiModels
+            .groupBy { it.data.category }
+            .flatMap { entry ->
+                val list = mutableListOf<ListItem>()
+                list.add(ListItem.Header(entry.key))
+                list.addAll(entry.value.map { ListItem.Package(it) })
+                list
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/ShippingLabelServicePackageAdapter.kt:30]: For displaying purposes, this converts the list of packages into a list with category header(s) displayed above its packages.

**Review by anitaa1990 — APPROVED** [review]
This looks good @hafizrahman! Nice work 🥇 ! Let me know if this is good to merge and I can merge it 👍

**Comment by hafizrahman** [comment]
@anitaa1990 Thank you! This is good to go :)

**Pull request merged**